src/core/annotation.js

```diff
@@ -1285,6 +1298,14 @@ class WidgetAnnotation extends Annotation {
 class TextWidgetAnnotation extends WidgetAnnotation {
   constructor(params) {
     super(params);
+    if (params.defaultAppearanceData) {
```

Comment:
I think you already indicated this, but individual annotations might have their own appearance, so the global one should only be used as a fallback in that case.